### PR TITLE
restore pull request: https://github.com/google/EarlGrey/pull/646

### PR DESCRIPTION
**Pull request by ynzhang0509**
also fixed some test logic

### CODE REVIEW

File: EarlGrey/Additions/UIViewController+GREYAdditions.m [hunk 1]
```diff
@@ -138,18 +138,22 @@ - (void)greyswizzled_viewWillAppear:(BOOL)animated {
...
+#if !defined(__IPHONE_10_0) || (__IPHONE_OS_VERSION_MIN_REQUIRED < __IPHONE_10_0)
+        [coordinator notifyWhenInteractionEndsUsingBlock:contextBlock];
+#else
+        [coordinator notifyWhenInteractionChangesUsingBlock:contextBlock];
+#endif
       }
 
       GREYAppStateTrackerObject *object =
-          TRACK_STATE_FOR_OBJECT(kGREYPendingViewsToAppear, self);
+      TRACK_STATE_FOR_OBJECT(kGREYPendingViewsToAppear, self);
```
Comment (tirodkar) [inline EarlGrey/Additions/UIViewController+GREYAdditions.m]: Fix formatting.

File: EarlGrey/Additions/UIViewController+GREYAdditions.m [hunk 2]
```diff
@@ -179,14 +183,19 @@ - (void)greyswizzled_viewWillDisappear:(BOOL)animated {
...
-              GREYAppStateTrackerObject *object =
-                  objc_getAssociatedObject(self, @selector(greyswizzled_viewWillAppear:));
-              UNTRACK_STATE_FOR_OBJECT(kGREYPendingViewsToDisappear, object);
-            }
-          }];
+      void (^contextBlock)(id<UIViewControllerTransitionCoordinatorContext>) =
+      ^(id<UIViewControllerTransitionCoordinatorContext> context) {
+        if ([context isCancelled]) {
+          GREYAppStateTrackerObject *object =
+          objc_getAssociatedObject(self, @selector(greyswizzled_viewWillAppear:));
```
Comment (tirodkar) [inline EarlGrey/Additions/UIViewController+GREYAdditions.m]: Formatting is off.

File: Tests/FunctionalTests/TestRig/Sources/FTRAlertViewController.m [hunk 3]
```diff
@@ -144,18 +144,16 @@ - (IBAction)showStyledAlert:(id)sender {
...
-    }
+  if (buttonIndex == 1) {
+    UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Danger!"
+                                                    message:@"*zap*"
+                                                   delegate:nil
+                                          cancelButtonTitle:@"Roger"
+                                          otherButtonTitles:nil];
+    [alert show];
+  }
+  if (buttonIndex == 2) {
```
Comment (khandpur) [inline Tests/FunctionalTests/TestRig/Sources/FTRAlertViewController.m]: why even need this?

Comment (ynzhang0509) [reply Tests/FunctionalTests/TestRig/Sources/FTRAlertViewController.m]: Hmm yes it looks weird. Will remove it.

**Comment by ynzhang0509** [comment]
Checked failures. All test failures are about FTRLocalUIWebViewTest, which is unrelated to this PR.

**Pull request merged**